SonarQube.TeamBuild.Integration/CoverageReportDownloader.cs

```diff
@@ -32,17 +40,44 @@ public bool DownloadReport(string reportUrl, string newFullFileName, ILogger log
             string downloadDir = Path.GetDirectoryName(newFullFileName);
             Utilities.EnsureDirectoryExists(downloadDir, logger);
 
-            using (WebClient myWebClient = new WebClient())
-            {
-                myWebClient.UseDefaultCredentials = true;
+            InternalDownloadReport(tfsUri, reportUrl, newFullFileName, logger);
+
+            return true;
+
+        }
 
-                logger.LogMessage(Resources.DOWN_DIAG_DownloadCoverageReportFromTo, reportUrl, newFullFileName);
+        private void InternalDownloadReport(string tfsUri, string reportUrl, string reportDestinationPath, ILogger logger)
+        {
+            VssCredentials vssCreds = null;
```

Comment:
Separate out the VSS-specific code into a separate method.
e.g. HttpMessageHandler GetHttpHandler(sting tfsUri)